plugin/src/main/java/com/playmonumenta/scriptedquests/quests/components/actions/dialog/DialogText.java

```diff
@@ -33,7 +33,12 @@ public DialogText(String displayName, JsonElement element) throws Exception {
 	@Override
 	public void sendDialog(Plugin plugin, Player player, Entity npcEntity, QuestPrerequisites prereqs) {
 		for (String text : mText) {
-			MessagingUtils.sendNPCMessage(player, mDisplayName, text);
+			if (mDisplayName != null && !mDisplayName.equals("")) {
```

Comment:
Try this:
https://docs.oracle.com/javase/10/docs/api/java/lang/String.html#isEmpty()

Reply:
Also note that the displayname falls back to the NPC's name - though I haven't read far enough down to see if this matches what you've got below.